catalog/spatial/geowebcache/geowebcache-admin-plugin/src/main/webapp/js/model/GwcLayers.js

```diff
@@ -0,0 +1,157 @@
+                    this.get("layers").push(layer);
+                } else {
+                    var updateLayer = this.getLayer(name);
+                    updateLayer.xmlUrl = xmlUrl;
+                    updateLayer.layerUrl = layerUrl;
+                    updateLayer.mimeTypes = mimeTypes;
+                    updateLayer.xmlLayers = xmlLayersArray;
+                }
+                this.trigger('change:layers', this);
+            },
+            parseLayersXml: function(data) {
+                var that = this;
+                $(data).find("layer").each(function () {
+                    var name = $(this).find("name").text();
+                    var url = "../../geowebcache/rest/layers/" + name + ".xml";
+                    $.ajax({
+                        url: url,
+                        dataType: 'xml',
+                        success: function(data) {
+                            that.parseLayerXml(name, data);
+                        }
+                    });
+                });
+            },
+            deleteLayer: function(name) {
+                var url = "../../geowebcache/rest/layers/" + name + ".xml";
+                var that = this;
+                $.ajax({
+                    url: url,
+                    type : 'DELETE',
+                    success: function() {
+                        var layers = that.get("layers");
+                        $.each(layers, function() {
+                            if(name === this.name) {
+                                layers.splice( $.inArray(this, layers), 1 );
+                                that.set("layers", layers);
+                                that.trigger('change:layers', this);
+                            }
+                        });
+                    }
+                });
+            },
+            postLayerData: function(layerData, type) {
+                var restUrl = "../../geowebcache/rest/layers/" + layerData.name + ".xml";
+                var that = this;
+                var data = this.generateXmlForLayer(layerData);
+                $.ajax({
+                    url: restUrl,
+                    type : type,
+                    dataType : "xml",
+                    data : data,
+                    success: function() {
+                        that.getLayers();
+                    }
+                });
+            },
+            addLayer: function(layerData) {
+                this.postLayerData(layerData, "PUT");
+            },
+            updateLayer: function(layerData) {
+                if(this.nameExistsInArray(layerData.name)) {
+                    this.postLayerData(layerData, "POST");
+                } else {
+                    this.deleteLayer(layerData.oldName);
+                    this.postLayerData(layerData, "PUT");
+                }
+            },
+            nameExistsInArray: function(name) {
+                return this.get("layers").some(elem => elem.name === name);
```

Comment:
Actually, I'm not sure that we can use arrow functions :disappointed:. They're not supported in IE or Safari: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Browser_compatibility